app/src/main/java/io/homeassistant/companion/android/bluetooth/ble/IBeaconMonitor.kt

```diff
@@ -82,4 +91,14 @@ class IBeaconMonitor {
         sensorManager.updateBeaconMonitoringSensor(context)
         SensorReceiver.updateAllSensors(context)
     }
+
+    public fun setUUIDFilter(uuidFilter: List<String>, uuidFilterExclude: Boolean) {
+        this.uuidFilter = uuidFilter
+        this.uuidFilterExclude = uuidFilterExclude
+
+        // existing beacons are only filtered when the filter changes
+        beacons
+            .filter { beacon -> ignoreBeacon(beacon.uuid) }
```

Comment:
This will return a new list, ie the original `beacons` is not updated. As you're not doing anything with the result I think you want to update the original list?

Reply:
Indeed, but the new list will contain references to the same objects as the old list, so it will modify the `skippedUpdated` field of those objects. I could do it with loop/if:
```
for (beacon in beacons)
    if (ignoreBeacon(beacon.uuid))
        beacon.skippedUpdated = MAX_SKIPPED_UPDATED
```
but I thought the "higher-order" (filter/forEach) style is a bit cleaner:
```
beacons
    .filter { ignoreBeacon(it.uuid) }
    .forEach { it.skippedUpdated = MAX_SKIPPED_UPDATED }
```